libafl_frida/src/alloc.rs

```diff
@@ -532,6 +498,49 @@ impl Allocator {
         map_to_shadow!(self, start)
     }
 
+    /// Checks whether the given address up till size is valid unpoisoned shadow memory.
+    #[inline]
+    #[must_use]
+    pub fn check_shadow(&self, address: *const c_void, size: usize) -> bool {
```

Comment:
When I was trying to debug some other issues, I used this function, vice the assembly one, as it was easier to debug.  I had issues with some of the corner cases, but am not sure if my fixes were correct.

- If address isn't aligned to 16 bits, and size < 8, shadow_size underflows.
- I think the checks for shadow_addrs need to be inverted?  For example, if there are 4 bytes at the end of an allocation, unpoison fills them with 0xf0, while the checks here compare that with 0x04 (on lines 514 and 535)